###########problem1################
library(ggplot2)
library(nycflights13)
library(patchwork)
flights
P1 <- ggplot(flights, aes(x=distance,y=arr_delay)) +
  geom_point() + 
  geom_smooth()

temp_flights <- flights %>%
  group_by(carrier) %>%
  summarize(m_arr_delay = mean(arr_delay,na.rm=TRUE))

P2 <- ggplot(temp_flights, aes(x=carrier,y= m_arr_delay)) +
  geom_bar(stat="identity")

P3 <- ggplot(flights, aes(x=carrier, y=arr_delay)) + 
  geom_boxplot()

P4 <- ggplot(flights, aes(x=arr_delay)) +
  geom_histogram()


png("patch.fig1.png", width = 7, height = 6, units="in", res = 600)
(P1 | P2) / (P3 | P4)
graphics.off()

Layout = '
AAAB
AAAC
AAAD
'
png("patch.fig2.png", width = 14, height = 12, units="in", res = 600)
P1 + P2 + P3 + P4 + plot_layout(design = Layout) + plot_annotation(tag_levels = 'A')
graphics.off()

Layout = '
AAAB
AAAC
AAAD
'
png("patch.fig3.png", width = 14, height = 12, units="in", res = 600)
P1 + P2 + P3 + P4 + plot_layout(design = Layout) + plot_annotation(tag_levels = 'A')
graphics.off()
###########problem2################
library("ATACseqQC")
library("Rsamtools")
bamfile = "ATAC.chrX3.bam"
# BAM file name
bamfile.labels = gsub(".bam", "", basename(bamfile))
# generate fragement size distribution
p4 = fragSizeDist(bamfile, bamfile.labels)

p5 = estimateLibComplexity(readsDupFreq(bamfile, index=bamfile))

## files will be output into outPath
outPath = "splited"
dir.create(outPath)

gal = readBamFile(bamfile, asMates=TRUE, bigFile=TRUE)
# shift the GAlignmentsLists by 5' ends.
# All reads aligning to the positive strand will be offset by +4bp,
# and all reads aligning to the negative strand will be offset -5bp
gal1 = shiftGAlignmentsList(gal)
shiftedBamfile = file.path(outPath, "shifted.bam")
# for reasons that are not clear (to me) BiocIO::export is not loaded above
library(BiocIO)
export(gal1, shiftedBamfile)


## split bam by fragment size (nucleosome free, mono, di-, etc)
library(TxDb.Dmelanogaster.UCSC.dm6.ensGene)
txs = transcripts(TxDb.Dmelanogaster.UCSC.dm6.ensGene)
TSS = promoters(txs, upstream=0, downstream=1)
TSS = unique(TSS)
objs = splitGAlignmentsByCut(gal1, txs = txs, outPath = outPath)
null = writeListOfGAlignments(objs, outPath)
## list the files generated by splitBam.
dir(outPath)

bamfiles = file.path(outPath, c("NucleosomeFree.bam","mononucleosome.bam","dinucleosome.bam","trinucleosome.bam"))
# estimate the library size for normalization
library(ChIPpeakAnno)
librarySize = estLibSize(bamfiles)
# Only consider these chromosomes (pick one for speed/memory)
seqlev = "chrX"
# seqlev = c("chrX","chr2L","chr2R","chr3L","chr3R")
NTILE = 101
dws = ups = 1010
sigs = enrichedFragments(bamfiles, TSS=TSS, librarySize=librarySize, seqlev="chrX", TSS.filter=0.5, n.tile = NTILE, upstream = ups, downstream = dws)

## log2 transformed signals
names(sigs) <- gsub(".bam", "", basename(names(sigs)))
sigs.log2 <- lapply(sigs, function(.ele) log2(.ele+1))
#plot heatmap
p6 <- featureAlignedHeatmap(sigs.log2, reCenterPeaks(TSS, width=ups+dws), zeroAt=.5, n.tile=NTILE)
library(gridExtra)
library(grid)
lay <- rbind(c(1,1,1,2,2,2),
             c(1,1,1,2,2,2),
             c(3,3,3,3,3,3))
png("problem5manhattan.png", width = 12, height = 6, units="in", res = 600)
grid.arrange(p6)
graphics.off()
library(patchwork)
png("test.png")
p4+p5+p6
graphics.off()


###########problem3################
install.packages("qqman")
library(qqman)
p1 <- read_tsv("problem1table.txt")
library(tidyverse)
p1 = p1 %>% mutate(SNP=str_c(chr,pos,sep = "_"))
levels(as.factor(p1$chr))
p1 = p1 %>% mutate(CHR = as.numeric(factor(chr)))
p1 = p1 %>% mutate(P = 10^-logp_interaction)
p1 = p1 %>% mutate(BP = pos)
p1.1 <- manhattan(p1, chr = "CHR", bp = "pos", p = "P", snp = "SNP", col = c("gray10", "gray60"),
          chrlabs = NULL,
          suggestiveline = -log10(1e-02),
          genomewideline = -log10(1e-01), annotatePval =TRUE)


p2 <- read_tsv("problem2table.txt")
library(tidyverse)
p2 = p2 %>% mutate(SNP=str_c(chr,pos,sep = "_"))
levels(as.factor(p2$chr))
p2 = p2 %>% mutate(CHR = as.numeric(factor(chr)))
p2 = p2 %>% mutate(P = 10^-logp_independent)
p2 = p2 %>% mutate(BP = pos)

p1.2 <- manhattan(p2, chr = "CHR", bp = "pos", p = "P", snp = "SNP", col = c("gray10", "gray60"),
          chrlabs = NULL,
          suggestiveline = -log10(1e-05),
          genomewideline = -log10(5e-08), annotatePval= TRUE)


#############problem4#################
source("myManhattanFunction.R")
p1.2 <- myManhattan(p1)
p2.2 <- myManhattan(p2)
#############problem5#################
p1$logp_independent <- p2$logp_independent
p3 <- ggplot(p1,aes(x=logp_interaction,y=logp_independent)) + geom_point(shape=19) +
  xlab("-logp_interaction") + ylab("-logp_independent")+geom_abline(slope = 1, intercept = 0)
library(gridExtra)
library(grid)
lay <- rbind(c(1,1,1,1,1,1,1,3,3,3,3,3,3),
             c(1,1,1,1,1,1,1,3,3,3,3,3,3),
             c(2,2,2,2,2,2,2,3,3,3,3,3,3),
             c(2,2,2,2,2,2,2,3,3,3,3,3,3))
png("problem5manhattan.png", width = 12, height = 6, units="in", res = 600)
grid.arrange(p1.2,p2.2,p3, layout_matrix = lay)
graphics.off()
